scripts/autoregress.py

```diff
@@ -15,301 +15,314 @@
 import ConfigParser
 import psycopg2
 
-# Change to True to generate known good files from a reference version of
-# MapReady.
-gen_refs = False
 
 def main():
-	(args, dirs) = get_arguments()
-	(clean, tmpdir, mapready, diffimage, diffmeta, log) =\
-			get_config_options(args.config)
-	tmpdir = os.path.abspath(os.path.expandvars(tmpdir))
-	mapready = os.path.expandvars(mapready)
-	diffimage = os.path.expandvars(diffimage)
-	diffmeta = os.path.expandvars(diffmeta)
-	log = os.path.abspath(os.path.expandvars(log))
-	logger = logging.getLogger(__name__)
-	verbosity = 1
-	if args.quiet:
-		verbosity = 0
-	elif args.debug:
-		verbosity = 2
-	if verbosity == 2:
-		logger.setLevel(logging.DEBUG)
-	elif verbosity == 0:
-		logger.setLevel(logging.ERROR)
-	elif verbosity == 1:
-		logger.setLevel(logging.INFO)
-	if args.output == "-":
-		logger.addHandler(logging.StreamHandler(sys.stdout))
-	elif args.output != None:
-		logger.addHandler(logging.FileHandler(args.output))
-	else:
-		logger.addHandler(logging.handlers.SysLogHandler(log))
-	if len(dirs) == 1:
-		workdir = os.path.abspath(dirs[0])
-	elif len(dirs) == 0:
-		workdir = os.getcwd()
-	logger.debug("tmpdir = {0}".format(tmpdir))
-	logger.debug("workdir = {0}".format(workdir))
-	if os.path.exists(tmpdir):
-		logger.warn("directory {0} exists, deleting contents".format(
-				tmpdir))
-		shutil.rmtree(tmpdir)
-	logger.debug("creating directory {0}".format(tmpdir))
-	os.mkdir(tmpdir)
-	conn = get_db(args.config)
-	if conn:
-		cur = conn.cursor()
-	failures = display_results(autoregress(workdir, tmpdir,
-			clean, (mapready, diffimage, diffmeta), cur))
-	if conn:
-		conn.commit()
-		cur.close()
-		conn.close()
-	if clean:
-		logger.debug("cleaning up")
-		os.rmdir(tmpdir)
-	sys.exit(failures)
+        (args, dirs) = get_arguments()
+        (tmpdir, mapready, diffimage, diffmeta, log) =\
+                        get_config_options(args.config)
+        tmpdir = os.path.abspath(os.path.expandvars(tmpdir))
+        mapready = os.path.expandvars(mapready)
+        diffimage = os.path.expandvars(diffimage)
+        diffmeta = os.path.expandvars(diffmeta)
+        log = os.path.abspath(os.path.expandvars(log))
+        logger = logging.getLogger(__name__)
```

Comment:
Read this about logging exceptions in the log file!
http://blog.tplus1.com/blog/2012/08/05/python-log-uncaught-exceptions-with-sys-excepthook/